test/widgets/all_channels_test.dart

```diff
@@ -202,21 +203,33 @@ void main() {
       } else {
         check(maybeToggle).isNull();
       }
-
-      check(findInRow(find.byIcon(ZulipIcons.more_horizontal))).findsOne();
     }
   });
 
-  testWidgets('tapping three-dots button opens channel action sheet', (tester) async {
+  testWidgets('open channel action sheet on long press', (tester) async {
     await setupAllChannelsPage(tester, channels: [eg.stream()]);
 
-    await tester.tap(find.byIcon(ZulipIcons.more_horizontal));
-    await tester.pump();
+    await tester.longPress(find.byType(AllChannelsListEntry));
+    await tester.pumpAndSettle();
     await transitionDurationObserver.pumpPastTransition(tester);
 
     check(find.byType(BottomSheet)).findsOne();
   });
 
+  testWidgets('navigate to channel feed on tap', (tester) async {
+    final channel = eg.stream(name: 'some-channel');
+    await setupAllChannelsPage(tester, channels: [channel]);
+    connection.prepare(json: eg.newestGetMessagesResult(
+      foundOldest: true, messages: [eg.streamMessage(stream: channel)]).toJson());
+
+    await tester.tap(find.byType(AllChannelsListEntry).at(0));
```

Comment:
nit: put the setup step `connection.prepare` immediately before the step that consumes it:
```suggestion
    await setupAllChannelsPage(tester, channels: [channel]);

    connection.prepare(json: eg.newestGetMessagesResult(
      foundOldest: true, messages: [eg.streamMessage(stream: channel)]).toJson());
    await tester.tap(find.byType(AllChannelsListEntry).at(0));
```